Mathlib/Analysis/SpecialFunctions/ImproperIntegrals.lean

```diff
@@ -81,6 +81,10 @@ theorem integrableOn_Ioi_rpow_iff {s t : ℝ} (ht : 0 < t) :
     exact Real.rpow_le_rpow_of_exponent_le x_one h
   exact not_IntegrableOn_Ioi_inv this
 
+theorem integrableAtFilter_rpow_atTop {r : ℝ} (hr : r < -1) :
```

Comment:
The converse of this is true as well, and might be useful. Can you add it? Here is a proof (might be golfable further)
```lean
theorem integrableAtFilter_rpow_atTop_iff {r : ℝ} :
    IntegrableAtFilter (fun x : ℝ ↦ x ^ r) atTop ↔ r < -1 := by
  refine ⟨fun h ↦ ?_, 
    fun hr ↦ ⟨Set.Ioi 1, Ioi_mem_atTop 1, (integrableOn_Ioi_rpow_iff zero_lt_one).mpr hr⟩⟩
  obtain ⟨s, hs, hr⟩ := h
  obtain ⟨t, ht⟩ := mem_atTop_sets.mp hs
  rw [← integrableOn_Ioi_rpow_iff (t := max t 1)]
  · exact hr.mono_set <| fun x hx ↦ ht _ <| (le_max_left t 1).trans hx.le
  · exact zero_lt_one.trans_le (le_max_right _ _)
```

Reply:
Do you think something like the following should be added to Mathlib (with the suitable hypothesis): 
```
theorem integrableAtFilter_atTop_iff {α E : Type*} (f : α → E) (μ : Measure α := by volume_tac) :
    IntegrableAtFilter f atTop μ ↔ ∃ a, IntegrableOn f (Ioi a) μ := by
```
or maybe
```
theorem integrableAtFilter_atTop_iff {α E : Type*} (f : α → E) (μ : Measure α := by volume_tac) :
    IntegrableAtFilter f atTop μ ↔ ∀ᶠ a in atTop, IntegrableOn f (Ioi a) μ
```
I don't think that would make the proof of the `iff` statement shorter though.

Reply:
Yes, I think we should definitely add one or the other; and I think the former is probably easier to apply in practice.

Reply:
#21016.  

Still, I don't think that would be useful here. So I kept your proof since I couldn't do better :) Thanks!